src/zarr/storage/_builtin_adapters.py

```diff
@@ -0,0 +1,562 @@
+"""
+Built-in store adapters for ZEP 8 URL syntax.
+
+This module provides store adapters for common store types that are
+built into zarr-python.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Literal, cast
+from urllib.parse import parse_qs, urlparse
+
+from zarr.abc.store_adapter import StoreAdapter
+from zarr.storage._local import LocalStore
+from zarr.storage._logging import LoggingStore
+from zarr.storage._memory import MemoryStore
+from zarr.storage._zep8 import URLStoreResolver
+from zarr.storage._zip import ZipStore
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from zarr.abc.store import Store
+    from zarr.abc.store_adapter import URLSegment
+
+__all__ = [
+    "FileSystemAdapter",
+    "HttpsAdapter",
+    "LoggingAdapter",
+    "MemoryAdapter",
+    "RemoteAdapter",
+    "S3Adapter",
+    "ZipAdapter",
+]
+
+
+class FileSystemAdapter(StoreAdapter):
+    """Store adapter for local filesystem access."""
+
+    adapter_name = "file"
+
+    @classmethod
+    async def from_url_segment(
+        cls,
+        segment: URLSegment,
+        preceding_url: str,
+        **kwargs: Any,
+    ) -> Store:
+        """Create a LocalStore from a file URL segment."""
+        # For file scheme, the preceding_url should be the full file: URL
+        if not preceding_url.startswith("file:"):
+            raise ValueError(f"Expected file: URL, got: {preceding_url}")
+
+        # Extract path from preceding URL
+        path = preceding_url[5:]  # Remove 'file:' prefix
+        if not path:
+            path = "."
```

Comment:
> For security, implementation SHOULD place limits on where this scheme is permitted.

refering to `..` which seems important here to not expose unlimited things in the filesystem.

Reply:
not really sure how to get into the business of defining what is safe here. But flagging for consideration